packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1728,19 +2161,24 @@ class Navigator extends StatefulWidget {
 //   route entry will exit that state.
 // # These states await futures or other events, then transition automatically.
 enum _RouteLifecycle {
-  // routes that are and will be present:
-  add, // we'll want to run install, didAdd, etc; a route created by onGenerateInitialRoutes or by the initial widget.pages
```

Comment:
yes, however add/adding is not considered present after page api. In Navigator we need to determine:
1, which route is considered present to user. This is willBePresent. This is used when annoucing previous route after multiple pop or decide if a route can receive user gesture.
2, which route is considered present to navigator. This is isPresent. The Navigator use this decide which route still worth notifying navigation change
3, which route is still animatable, whether or not navigator cares about them. This is canTransit. These routes still need to be chained up with didchangeprev and didchangenext so the secondary animations are still wired it up.

When a page is add/adding, it is waiting for the top most route push to finish. In this case, this page is consider not exist in the app yet and not suitable for transition as well. Thus it is rated below removing

Reply:
`add` will go to `adding` synchronically. If there is no top pushing route, it will go to `idle` synchronically. If a route is `add` or `adding` it means it is waiting for top pushing route and should not be displayed and thus non transition-able. 

it will actually create some issue if we consider it transition-able

Consider this case, we have routes [A, B, C]. If we put D, E, F on top, it become [A, B, C ,D(add), E(add), F(push)].
Navigator will try to announce F.didChangePrev(E)  and  E.didChangeNext(F) to hook up secondary animation because it think E is transition-able. This will result C's secondary animation never hooked up with F and will not transition with F

Reply:
talked offline, this will also affect some imperative api to determine whether a route can be popped. Need to refactor it